siaskynet/skynet.py

```diff
@@ -79,7 +79,7 @@ def upload_directory_request(path, opts=None):
         files = list(Skynet.walk_directory(path).keys())
         for file in files:
             ftuples.append((opts.portal_directory_file_fieldname,
-                            (file, open(file, 'rb'))))
+                            (file[len(path):], open(file, 'rb'))))
```

Comment:
Shouldn't it be `[:len(path)]` if you want to cut the initial path? Also, I would first assert that `file` starts with `path`.